### ISSUE TITLE
Can't compile right

### ISSUE THREAD

**Issue by kud** [halffullheart/mdr#36]
```
$ brew install mdr
==> Downloading https://github.com/halffullheart/mdr/archive/v1.0.0.tar.gz
######################################################################## 100.0%
==> rake
==> rake release
rake aborted!
Command failed with status (1): [gcc -arch i386 -isysroot /Applications/Xco...]
/private/tmp/mdr-c6Sg/mdr-1.0.0/Rakefile:90:in `block in <top (required)>'
Tasks: TOP => release => release/mdr
(See full trace by running task with --trace)
```

**Comment by MoOx** [comment]
Just got the same error here.

```
❯❯ ruby --version
ruby 2.0.0p451 (2014-02-24 revision 45167) [universal.x86_64-darwin13]

❯ rake --version
rake, version 0.9.6

❯❯ gcc --version
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 5.1 (clang-503.0.40) (based on LLVM 3.4svn)
Target: x86_64-apple-darwin13.2.0
Thread model: posix
```

**Comment by zyegfryed** [comment]
Just got the same error, too. More detail on that [gist](https://gist.github.com/anonymous/e3a23dff9cce78ee5c20) generated with `brew gist-logs mdr`.
It might be an Xcode issue...

**Comment by kud** [comment]
I think everytime xcode will change the version, we'll be fucked here, no?

**Comment by zyegfryed** [comment]
FYI as a workaround I'm using the [binary](http://mac.softpedia.com/get/Developer-Tools/MDR-Make-Diffs-Readable.shtml) downloadable from Softpedia.

**Comment by zyegfryed** [comment]
@kud  can you elaborate? Have you fixed it in the past?

**Comment by kud** [comment]
The path of xcode is static, and doesn't handle the update of xcode version, like `/path/xcode/6.0.1/bin`

If xcode is updated, the path isn't the same anymore and mdr compilation will be broken.

Or maybe we talk about another issue...?

**Comment by zyegfryed** [comment]
@kud  I'm afraid I don't get it.
From the [gist I've posted](https://gist.github.com/anonymous/e3a23dff9cce78ee5c20) can you tell me which path is wrong according to you?

**Comment by kud** [comment]
`gcc -Wall -g -arch i386 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.8.sdk -mmacosx-version-min=10.8 -c Reader.c
clang: warning: no such sysroot directory: '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.8.sdk'`

This one. It's not xcode version in fact, it's macosx version. We're now on `macosx10.10.sdk` and i'm not sure you've got `MacOSX10.8.sdk` (well I don't have it)

**Comment by zyegfryed** [comment]
Indeed. Thanks having spotted it for me.

**Comment by zyegfryed** [comment]
Until it's fixed, a workaround is to symlink the targeted SDK, like so:

```
$  ln -s /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.{10,8}.sdk
```

**Comment by kud** [comment]
lol... yes, indeed. :)

**Comment by halffullheart** [comment]
Does anyone know if there is a way to detect the SDK path so it doesn’t have to be hardcoded? @kud is right this is just going to keep breaking.

**Comment by zyegfryed** [comment]
I gave a quick look at it and haven't found something probative. Sadly, I don't know ruby to help more. (I looked at RbConfig module, but haven't found something usable, yet.)
The idea would be to retrieve the OS version (like you can do with `lsb_release`, not available by default on Mac), and then use it to build the full Xcode path.
If you know how to do that in ruby, just shoot!

**Comment by halffullheart** [comment]
Looks like this can be done by parsing the output of  `xcodebuild -sdk -version`. I’m working on a fix now.

**Comment by halffullheart** [comment]
Fixed by 577e757f9a2c47e211df4518de7164d4102a9a1d
